File: microlearning-adhd-app/backend/app/models/quiz.py
from datetime import datetime

from sqlmodel import Field, SQLModel, UniqueConstraint


class QuizSubmission(SQLModel, table=True):
    __table_args__ = (
        UniqueConstraint(
            "participant_id",
            "topic_id",
            "attempt",
            name="uq_quiz_submission_participant_topic_attempt",
        ),
    )

    id: int | None = Field(default=None, primary_key=True)
    participant_id: str = Field(foreign_key="participantsession.id")
    assignment: str
    subgroup: str
    video_id: str | None = None
    video_index: int | None = None
    topic_id: str
    answers_json: str
    answer_count: int
    attempt: int = Field(default=1)
    submitted_at: datetime

File: microlearning-adhd-app/frontend/src/services/types/quiz.ts
import { type GroupAssignment, type Subgroup } from './../../utils/groupAssignment'

export type QuizSubmission = {
  participant_id: string
  answer_count: number
  submitted_at: string
}

export type QuizAnswerSubmission = {
  assignment: GroupAssignment
  subgroup: Subgroup
  video_id: string | null
  video_index: number | null
  topic_id: string
  answers: Record<string, string[]>
  attempt?: number | null
}
